src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1419,6 +1419,11 @@ protected virtual void OnRequestTransmissionAborted(JsonRpcRequest request)
 
                     JsonRpcMethodAttribute? attribute = mapping.FindAttribute(method);
 
+                    if (attribute == null && options.UseSingleObjectParameterDeserialization)
+                    {
+                        attribute = new JsonRpcMethodAttribute(null) { UseSingleObjectParameterDeserialization = true };
```

Comment:
Synthesizing an attribute is fine, but the way you're doing it you break the code on line 1438 that checks whether `attribute == null` to do other default things.
Maybe we need a boolean local variable to track whether the attribute originally existed.

Reply:
Are you sure we should check for the existence of the attribute (line 1438) and not the explicit name inside the attribute itself? If i add an attribute (with name = null) to a method, should it still be possible to call a method by omitting the async suffix?

Reply:
What would you check within the existing attribute itself? The property you added is just a `bool` property, so it will always be `false` unless it was set to `true` explicitly.
I think your current implementation that only sets the value to `true` if the attribute is entirely absent makes sense. Your original problem statement was that you wanted to avoid adding the attribute to every method, which is fair. But if the attribute is *already* on it, considering that method to be explicitly defined and not inherit defaults from elsewhere is a nice, simple model.

Reply:
My original issue is solved. My point is not related to my original issue (but kind of to the solution of this bug).

Lets assume somebody defines this method:
```csharp
public async Task<string> DoSomethingAsync()
```
you can now call this method with an rpc call `DoSomething` and `DoSomethingAsync` (is this correct?)

Later, somebody adds an single object parameter like this:
```csharp
[JsonRpcMethod(UseSingleObjectParameterDeserialization = true)]
public async Task<string> DoSomethingAsync(SomeInputData input)
```
Since there is an attribute now, you can no longer call this method via `DoSomething` (even though the developer did change anything about the naming). Is this intended?

My solution would be to change line 1438 from this:
```csharp
if (attribute == null && method.Name.EndsWith(ImpliedMethodNameAsyncSuffix, StringComparison.Ordinal))
```
to this:
```csharp
if (attribute?.Name == null && method.Name.EndsWith(ImpliedMethodNameAsyncSuffix, StringComparison.Ordinal))
```

Let me know if i am understanding something wrong.